src/OpenTK.Windowing.Desktop/NativeWindow.cs

```diff
@@ -1151,6 +1304,24 @@ public virtual void ProcessEvents()
             {
                 GLFW.PollEvents();
             }
+
+            RethrowCallbackExceptionsIfNeeded();
+        }
+
+        private void RethrowCallbackExceptionsIfNeeded()
+        {
+            if (_callbackExceptions.Count == 1)
+            {
+                Exception exception = _callbackExceptions[0];
+                _callbackExceptions.Clear();
+                throw exception;
```

Comment:
You may want to use `ExceptionDispatchInfo.Capture(exception).Throw();` here instead so that the exception gets rethrown without bulldozing the stack trace (IIRC?) [Example](https://github.com/space-wizards/RobustToolbox/blob/c4062bcae95e965301c22768653b581287253763/Robust.Client/Graphics/Clyde/Clyde.Windowing.cs#L599)

Another thing to consider is maybe always wrapping the exception in something similar to `TargetInvocationException`.